CountryPickeriOS13+/CountryPickerSPM/Sources/CountryPickerSPM/CountryManager.swift

```diff
@@ -0,0 +1,242 @@
+//
+//  CountryManager.swift
+//  CountryCodeInSwift3
+//
+//  Created by Cl-macmini-100 on 12/19/16.
+//  Copyright © 2016 Suryakant. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+// MARK: - CountryManagerInterface
+public protocol CountryListDataSource: ObservableObject {
+    func country(withCode code: String) -> Country?
+    func allCountries(_ favoriteCountriesLocaleIdentifiers: [String]) -> [Country]
+    var lastCountrySelected: Country? { get set }
+    func filterCountries(searchText: String) -> [Country]
+}
+
+// MARK: - CountryManagerInterface extension for optional variables and default implementation
+extension CountryListDataSource {
+    var lastCountrySelected: Country? {
+        get { nil }
+        set {}
+    }
+}
+
+// MARK: - CountryFilterOption
+/// Country filtering options
+public enum CountryFilterOption {
+    /// Filter countries by country name
+    case countryName
+    
+    /// Filter countries by country code
+    case countryCode
+    
+    /// Filter countries by country dial code
+    case countryDialCode
+}
+
+#if SWIFT_PACKAGE
+let bundle = Bundle.module
+#else
+let bundle = Bundle(for: Country.self)
+#endif
+
+// MARK: - CountryManager
+open class CountryManager: CountryListDataSource {
+    
+    // MARK: - Variables
+    @Published public var countries = [Country]()
+    
+    private var countriesFilePath: String? {
+        let countriesPath = bundle.path(forResource: "CountryPickerController.bundle/countries", ofType: "plist")
+        return countriesPath
+    }
+    
+    public static let shared: CountryManager = {
+        let countryManager = CountryManager()
+        do {
+            try countryManager.loadCountries()
+        } catch {
+            #if DEBUG
+            print(error.localizedDescription)
+            #endif
+        }
+        return countryManager
+    }()
+    
+    /// Current country returns the country object from Phone/Simulator locale
+    open var currentCountry: Country? {
+        guard let countryCode = Locale.current.regionCode else {
+            return nil
+        }
+        return Country(countryCode: countryCode)
+    }
+    
+    public var lastCountrySelected: Country?
```

Comment:
We should make getters in this class serially accessed otherwise we might face threading issues.